trillian/migrillian/election/election.go

```diff
@@ -22,36 +22,43 @@ import "context"
 // Election controls an instance's participation in master election process.
 // Note: Implementations are not intended to be thread-safe.
 type Election interface {
-	// Await blocks until the instance captures mastership. Returns a "mastership
-	// context" which remains active until the instance stops being the master,
-	// or the passed in context is canceled. Returns an error if capturing fails,
-	// or the passed in context is canceled before mastership is captured.
-	//
-	// Await is safe to be called again when the current mastership context is
-	// canceled, which might happen implicitly when mastership is overtaken, or
-	// explicitly when Resign is called.
+	// Await blocks until the instance captures mastership. Returns immediately
+	// if it is already the master. Returns an error if capturing fails, or the
+	// passed in context is canceled before mastership is captured. If an error
+	// is returned, the instance might still have become the master. Idempotent,
+	// might be useful to retry in case of an error.
+	Await(ctx context.Context) error
+
+	// Observe returns a "mastership context" which remains active until the
+	// instance stops being the master, or the passed in context is canceled. If
+	// the instance is not the master during this call, returns an outright
+	// canceled context.
 	//
-	// TODO(pavelkalinnikov): It makes sense to distinguish the ctx being passed
-	// in only for waiting, and a "master context" used to derive the mastership
-	// context from. This way, we could put a deadline on waiting for mastership,
-	// and not cancel mastership in case it gets acquired within the deadline.
-	Await(ctx context.Context) (context.Context, error)
+	// The resources used for mainaining the mastership context are released when
+	// the latter gets canceled. This happens when the instance loses / gives up on
```

Comment:
By "gives up on mastership" do you mean "calls Resign()" ?

Reply:
Yes, reworded.